src/mono/wasm/debugger/BrowserDebugProxy/EvaluateExpression.cs

```diff
@@ -445,12 +449,33 @@ private static async Task<IList<JObject>> ResolveElementAccess(ExpressionSyntaxR
 
                 syntaxTree = replacer.ReplaceVars(syntaxTree, null, null, null, elementAccessValues);
             }
-
             expressionTree = syntaxTree.GetCompilationUnitRoot(token);
             if (expressionTree == null)
                 throw new Exception($"BUG: Unable to evaluate {expression}, could not get expression from the syntax tree");
+            var variableDef = await GetVariableDefinitions(resolver, replacer.variableDefinitions, replacer.hasExpressionStatement, token);
+            return await EvaluateSimpleExpression(resolver, syntaxTree.ToString(), expression, variableDef, logger, token);
+        }
 
-            return await EvaluateSimpleExpression(resolver, syntaxTree.ToString(), expression, replacer.variableDefinitions, logger, token);
+        internal static async Task<List<string>> GetVariableDefinitions(MemberReferenceResolver resolver, List<(string idName, JObject obj, string definition)> variableDefinitions, bool callToString, CancellationToken token)
+        {
+            var variableDef = new List<string>();
+            foreach ( var definition in variableDefinitions )
+            {
+                if (!callToString || definition.obj?["type"]?.Value<string>() != "object")
```

Comment:
What does `callToString` mean? should it be `canCallToString`, or `hasToString`?

Reply:
It means that it should call toString even if it's not overriden in the class.